Sources/SwiftSyntax/Syntax.swift

```diff
@@ -323,7 +323,23 @@ public extension SyntaxProtocol {
   /// Returns a new syntax node with its leading trivia replaced
   /// by the provided trivia.
   func withLeadingTrivia(_ leadingTrivia: Trivia) -> Self {
-    return Self(Syntax(data.withLeadingTrivia(leadingTrivia)))!
```

Comment:
@ahoppen this is my first take. 
Not sure if there is a better place to do this manipulation? 


And we should properly also add it, so it works with trailing trivia.

Reply:
I don’t think we should make this kind of manipulation in the SwiftSyntax module. My take so far has been that SwiftSyntax itself (and the `SyntaxFactory` methods) should allow you to generate any Swift source code and don’t do anything that you might want to call “obvious fix-ups”. So I think the place where this belongs is where we call `withLeadingTrivia` in `SwiftSyntaxBuilder`.